lib/any.js

```diff
@@ -683,6 +683,21 @@ module.exports = internals.Any = class {
             }
         }
 
+        if (this._settings) {
+
+            const optionSet = ['abortEarly', 'convert', 'allowUnknown', 'skipFunctions', 'stripUnknown', 'language', 'presence', 'context', 'noDefaults'];
+            if (optionSet.some((option) => this._settings.hasOwnProperty(option))) {
```

Comment:
aren't doing a double check on `this._settings.hasOwnProperty`? I assume this is to prevent setting the empty object should https://github.com/hapijs/joi/pull/1132/files#diff-834068731e2e6ef966ecfef994bbae5eR694 always be false?